Conda instructions

RemiLehe opened the pull request:
I recently uploaded openPMD-viewer to Anaconda.org: see [here](https://anaconda.org/rlehe/openpmd_viewer). 

Therefore, the current pull request introduces two changes:

- The installation documentation has been updated, so as to explain how to install `openPMD-viewer` with `conda`
- The instructions to upload future releases to Anaconda.org have been added. Note that the instructions for releasing used to be in `CONTRIBUTING.md`. But since this was not really appropriate and since `CONTRIBUTING.md` started to be quite long, I moved the instructions on releasing into a new file `RELEASING.md`.

@ax3l I thought you might be interested in reviewing this PR and in offering advice on the release process. But if you are too busy, just let me know.

RemiLehe commented:
NB: I also streamlined and improved a bit the conda recipe (`meta.yml` file)

ax3l commented:
@RemiLehe thank you for the update!
I think moving the release instructions to its own file makes sense since CONTRIBUTING is for (external) developers and releasing is a maintainer task which is not important if one just wants to contribute code/fixes/etc.

Have you thought about adding your recipe to [conda forge](https://conda-forge.github.io/)? That looks like a great place for continuously delivering our releases.

I am also planning to add a [spack receipe](https://github.com/LLNL/spack) since building HDF5 (even parallel) works like a charm with it and it does not ship binary blobs (like conda) but actual auto-build instructions (like pip).

ax3l commented on RELEASING.md:
```diff
@@ -0,0 +1,84 @@
+# Creating a new release
+
+This document explains how to create a new release. In future
+versions of this packages, some of the steps below will be automatized.
```
Add: `This document is only relevant for maintainers of openPMD-viewer.`

ax3l commented on RELEASING.md:
````diff
@@ -0,0 +1,84 @@
+git push
+```
+
+- Create a new release through the graphical interface on Github
````
I actually just found out that tagging locally is even better, since you can sign your releases with your gpg key. One can then still draft a release with the GitHub API for the `git push --tags` one just uploaded (which is just an alias for the sha again).

We can keep that in mind, but I will personally start to sign all upcoming releases I maintain at HZDR.

ax3l commented on RELEASING.md:
````diff
@@ -0,0 +1,84 @@
+conda convert $(conda build --python=3.5 . --output) -p all
+```
+
+- Upload the differents versions to Anaconda.org
````
little typo: different versions

ax3l commented on RELEASING.md:
````diff
@@ -0,0 +1,84 @@
+anaconda upload osx-64/*
+anaconda upload linux-64/*
+anaconda upload linux-32/*
+```
````
and win? :D

RemiLehe replied:
what about it?

ax3l replied:
just curious if it works on win.

RemiLehe replied:
Yes, I think you can do `anaconda upload win-64/*`. But I am not sure it would work out of the box, and also I wouldn't now how to test it.

ax3l replied:
I am actually suprised how conda builds a linux and an osx version on the very same box. do you know?

ax3l commented on conda_recipe/meta.yaml:
```diff
@@ -23,4 +34,5 @@ test:
-  license_file: LICENSE.txt
\ No newline at end of file
+  license_file: LICENSE.txt
+  summary: "Visualization tools for openPMD files"
```
just a minor: pls add an empy newline to this file

ax3l replied:
added

ax3l commented:
That's a wonderful PR, thank you so much. The ` RELEASING.md` is the description I ever wanted without omitting the configs "assumed to be already in place". Wuhu! :sparkles: 

I added a few very minor inline comments, everything else looks great! :rocket:

ax3l commented on RELEASING.md:
````diff
@@ -0,0 +1,84 @@
+```
+rm -rf dist
+python setup.py sdist bdist_wheel
+twine upload dist/* -r pypi
````
does there an equivalent of `python setup.py sdist upload --sign` exist to sign an upload with gpg?

RemiLehe commented:
Thanks for reviewing, and for the very interesting comments.
I updated the PR and fixed the typos you spotted.

Regarding conda-forge and gpg signing: these are definitely two things I'd like to try. But I'll need a bit more time to read about it, so I guess this will be in a future, separate PR.

RemiLehe commented:
Could you merge whenever possible, if the above is fine? I'd like to do a new release soon, since it turns out that the current release has a problem with Python 3 and pip, which I recently fixed on `dev`.

Thanks!

ax3l commented:
absolutely good to go! :ship:

RemiLehe commented:
Thank you !

ax3l commented:
thank you for the conda work! :)
